packages/webpack-config/src/webpackConfig.ts

```diff
@@ -0,0 +1,636 @@
+        entryOnly: true,
+        raw: true,
+        test: "runtime.js",
+      }),
+    );
+  }
+
+  return extendWebpackConfig({
+    mode,
+    name: settings.name,
+    cache,
+    context: fs.realpathSync(process.cwd()), // follow any symlinks to ensure sourcemap path accuracy
+    devServer,
+    devtool: release ? "source-map" : "eval-source-map",
+    entry,
+    module: {
+      rules,
+      noParse: settings.noParse,
+      // report missing exports as errors instead of warnings
+      strictExportPresence: true,
+    },
+    optimization: {
+      minimizer: [
+        new TerserPlugin({
+          parallel: true,
+          // disable mangling for better debugging support
+          // in production and to reduce compilation time
+          terserOptions: { mangle: false },
+        }),
+        new CssMinimizerPlugin(),
+      ],
+      runtimeChunk: "single", // allows multiple entry points to be used on a page
+      splitChunks: {
+        chunks: "all", // extract common dependencies into shared bundles (JS and CSS)
+        maxAsyncRequests: Infinity,
+        maxInitialRequests: Infinity,
+      },
+    },
+    output: {
+      devtoolModuleFilenameTemplate: (info: { resourcePath: string }) => {
+        const cleanResourcePath = info.resourcePath.replace(/^\.\//, ""); // strip leading "./"
+        return `webpack://${settings.publicPathRoot ?? ""}/${cleanResourcePath}`;
+      },
+      filename: settings.outputFileName ?? "[name].js",
+      path: path.join(process.cwd(), settings.outputPath),
+      pathinfo: !release,
+      publicPath: isDevServer ? devServerPublicPath : virtualPublicPath,
+    },
+    plugins,
+    resolve: {
+      extensions: [".tsx", ".ts", ".js", ".json", ".css"],
+
+      // Do not follow symlinks when resolving modules.
+      // Without this, dependencies of *linked* modules
+      // would be resolved by walking up the real path of
+      // the module, rather than the linked path.
+      //
+      // For example:
+      // - <dev_root>\SamplePliWeb
+      // - <dev_root>\web-pli\dist -> (linked) <dev_root>\SamplePliWeb\src\P.S.W\node_modules\@pandell\classic
+      //
+      // Without the following setting, require("knockout")
+      // in pli/somemodule would be resolved to
+      // <dev_root>\web-pli\node_modules\knockout
+      // when it should be resolved to the copy in
+      // SamplePliWeb.
+      symlinks: false,
+    },
+    resolveLoader: {
+      modules: [
+        // resolve loaders from local node_modules to ensure loaders
+        // are found when this package is linked
+        path.join(__dirname, "..", "node_modules"),
+        // resolve loaders from web-pli root node_modules
+        path.join(__dirname, "..", "..", "..", "node_modules"),
+        "node_modules",
+      ],
+    },
+  } as WebpackConfigurationWithDevServer);
+}
+
+/**
+ * Creates webpack configuration function for the given settings.
+ */
+export function createWebpackConfig(
```

Comment:
Disclaimer: I only glanced at this code, I assume it is mostly a copy of `@pandell/cli` implementation. As I said in my other comment, webkit is definitely a dead-end, and the sooner we move off everywhere the better. Vite is a next-generation bundler that virtually all front-end frameworks have switched to by now.